test/unit/util/test_utils.py

```diff
@@ -149,6 +149,13 @@ class Stuff(str, Enum):
     "doi:10.1234567890/42",  # longer prefix
     "doi:10.1234/42ab:%&*$//crazy-suffix/%/&/",
     "doi:10.1234/aa",
+    "http://doi.org/10.1234/42",
+    "stillvalid:10.1234/42",
+    "dx.doi.org:10.1234/42",
+    "https://dx.doi.org:10.1234/42",
+    "doi:10.1234/42/a b",
```

Comment:
With the `.` in the regex, what would be the boundary separating the DOI from the context?

Reply:
none, space is allowed character in this part of doi 😭